pandas/core/indexes/datetimelike.py

```diff
@@ -854,6 +854,116 @@ def _concat_same_dtype(self, to_concat, name):
             new_data = np.concatenate([c.asi8 for c in to_concat])
         return self._simple_new(new_data, **attribs)
 
+    def _slice_by_value(self, start, end):
+        sliced = self.values[slice(*self.slice_locs(start, end))]
+        return self._shallow_copy(sliced)
+
+    def _intersect_ascending(self, other, **empty_params):
+        # to make our life easier, "sort" the two ranges
+        if self[0] <= other[0]:
+            left, right = self, other
+        else:
+            left, right = other, self
+
+        end = min(left[-1], right[-1])
+        start = right[0]
+
+        if end < start:
+            return type(self)(data=[], **empty_params)
+        return left._slice_by_value(start, end)
+
+    def _intersect_descending(self, other, **empty_params):
+        # this is essentially a flip of _intersect_ascending
+        if self[0] >= other[0]:
+            left, right = self, other
+        else:
+            left, right = other, self
+
+        start = min(left[0], right[0])
+        end = right[-1]
+
+        if end > start:
+            return type(self)(data=[], **empty_params)
```

Comment:
this is very clunky. again just return an indexer (or values are ok), and wrap at a higher level, then you don't need all of these params.